source/frontend/src/components/Table/Table.tsx

```diff
@@ -378,12 +383,11 @@ export const Table = <T extends IIdentifiedObject, TFilter extends object = {}>(
                         if (isSingleSelect === true) {
                           setExternalSelectedRows && setExternalSelectedRows([...values]);
                         } else {
-                          const allPreviouslySelected = instance.rows
-                            .filter(row => row.isSelected)
-                            .map(row => row.original);
+                          const allPreviouslySelected = selectedRowsRef?.current ?? [];
```

Comment:
This logic ensures that selected results on previous pages persist. This logic is in an arrow function so the value of externalSelectedRows is scoped out and will not update from the initial value ever. Using the ref here as a global in combination with the above useEffect ensures that each column has an up to date list of selected rows.